packages/dapp/src/actionCreators/network.ts

```diff
@@ -1,12 +1,24 @@
 import { AnyAction } from "redux";
+import { Subscription } from "rxjs";
 
 export enum networkActions {
   SET_NETWORK = "SET_NETWORK",
 }
 
-export const setNetwork = (network: string): AnyAction => {
+const internalSetNetwork = (network: string): AnyAction => {
   return {
     type: networkActions.SET_NETWORK,
     data: network,
   };
 };
+
+export const setNetwork = (network: string): any => {
+  return (dispatch: any, getState: any): AnyAction => {
+    const { listingHistorySubscriptions } = getState().networkDependent;
+    listingHistorySubscriptions.forEach((subscription: Subscription, key: string) => {
+      subscription.unsubscribe();
```

Comment:
should the listing also be removed from the map?

Reply:
when the SET_NETWORK action is processed in the network reducer, everything is cleared out, including this map